metering-billing/lambdas/src/main/java/com/amazon/aws/partners/saasfactory/metering/onboarding/BillingIntegration.java

```diff
@@ -271,58 +272,94 @@ private void provisionTenantInStripe(String tenantId, String planId) throws Stri
         }
 
-        final String customerId = "tenant-" + tenantId.split("-")[0];  //get first segment
-        LOGGER.info("provisionTenantInStripe: Setting up Tenant: " + customerId + " as a customer and creating subscription");
+        final String customerName = "tenant-" + tenantId.split("-")[0];  //get first segment
+        LOGGER.info("provisionTenantInStripe: Setting up Tenant: {} as a customer and creating subscription", customerName);
 
-        try {
-            //Only attempt to setup tenant if customer does not already exist for this tenant
-            try {
-                Customer customer = Customer.retrieve(customerId);
-                LOGGER.info("Customer : " + customerId + " already exists, setup of tenant: " + tenantId + " skipped");
+        Customer customer = getStripeCustomer(customerName);
+
+        //Only attempt to setup tenant if customer does not already exist for this tenant
+        if (null == customer) {
+            LOGGER.error("No existing customer: {} found in Stripe, so one will be created.", customerName);
+            if (planEnum == SubscriptionPlan.product_none)  {
+                LOGGER.info("Plan of {} for tenant {} will not be setup for billing.", planEnum.name(), tenantId);
                 return;
+            }
+            try {
+                CustomerCreateParams params =
+                        CustomerCreateParams.builder()
+                                .setName(customerName)
+                                //.setEmail("bogus@example.com")
+                                .setPaymentMethod("pm_card_visa")
+                                .setInvoiceSettings(
+                                        CustomerCreateParams.InvoiceSettings.builder()
+                                                .setDefaultPaymentMethod("pm_card_visa")
+                                                .build())
+                                .setDescription(customerName + " from AWS SaaS Boost")
+                                .build();
+                customer = Customer.create(params);
+                //create the subscription
+                createStripeSubscription(customer, planEnum, tenantId,false);
+                LOGGER.info("provisionTenantInStripe: Customer created with id: " + customer.getId());
+            } catch (StripeException e) {
+                LOGGER.error("provisionTenantInStripe: Error creating Customer for Tenant: " + tenantId);
+                LOGGER.error(Utils.getFullStackTrace(e));
+                throw e;
+            }
+        } else {
+            LOGGER.info("provisionTenantInStripe: Customer: {} create skipped, already exists with id: {} ", customerName, customer.getId());
+            //If subscription plan has changed then cancel the old subscriptions and create a new.
+            //Ideally, the subscription information for the tenant would be stored in a DDB table so we do not have to hit Stripe to find if it has changed
```

Comment:
If the customer can change subscription information directly in Stripe, we should *not* store any subscription info in DDB as that information may become inconsistent. We can build optimizations around Stripe (or other billing providers) if we find that it becomes a bottleneck.

Reply:
In this case by customer I meant the SaaS vendor/builder/ISV using Boost, as I would assume they have the highest level of control eventually over the Stripe subscription of behalf of the tenant; after all, the Stripe subscription is so the ISV can bill the tenant for usage.

I'm fine with sticking with this current implementation, I'm just suggesting storing information in DDB may have some unforeseen side effects.